### PR TITLE
[DROOLS-7598] Upgrade tomcat version

### PR DESCRIPTION
**Pull request by tkobayas**
**JIRA**: 
https://issues.redhat.com/browse/DROOLS-7598

**referenced Pull Requests**: 

* https://github.com/kiegroup/droolsjbpm-integration/pull/3007

You can check Kiegroup organization repositories CI status from [Chain Status webpage](https://kiegroup.github.io/droolsjbpm-build-bootstrap/)

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used locally on command line or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it. 

A general local execution could be the following one, where the tool clones all dependent projects starting from the `-sp` one and it locally applies the pull request (if it exists) in order to reproduce a complete build scenario for the provided *Pull Request*.

> **Note:** the tool considers multiple *Pull Requests* related to each other if their branches (generally in the forked repositories) have the same name.

``` shell
$ build-chain-action -df 'https://raw.githubusercontent.com/${GROUP:kiegroup}/droolsjbpm-build-bootstrap/${BRANCH:main}/.ci/pull-request-config.yaml' build pr -url <pull-request-url> -sp kiegroup/kie-wb-distributions [--skipExecution]
```

> Consider changing `kiegroup/kie-wb-distributions` with the correct starting project.

</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest</b> (using <i>this</i> e.g. <b>Jenkins retest this</b> optional but no longer required)
 
* for a <b>full downstream build</b> 
  * for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  * for <b>github actions</b> job: add the label `run_fdb`
    
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>

* <b>for windows-specific os job</b> add the label `windows_check`
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by tkobayas** [comment]
Jenkins run fdb

**Review by mariofusco — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -150,7 +150,7 @@
     <version.org.apache.neethi>3.1.1</version.org.apache.neethi>
     <version.org.apache.poi>4.1.2</version.org.apache.poi>
     <version.org.apache.sshd>2.9.2</version.org.apache.sshd>
-    <version.org.apache.tomcat>6.0.32</version.org.apache.tomcat>
+    <version.org.apache.tomcat>9.0.84</version.org.apache.tomcat>
```
Comment (mareknovotny) [inline pom.xml:153]: this is probably good to have latest, but i was also looking into matching the JWS which now uses 9.0.62, but i think we can try to use the latest, greatest ;)

**Review by mareknovotny — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (mareknovotny) [reply pom.xml:153]: the only problem now is we use/declare another version here https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/2413/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R280

**Review by mareknovotny — CHANGES_REQUESTED** [review]
please use only one version https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/2413/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R280

**Comment by tkobayas** [comment]
Jenkins run fdb

**Comment by tkobayas** [comment]
> please use only one version https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/2413/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R280

@mareknovotny Thanks, fixed.

**Review by mareknovotny — APPROVED** [review]

**Comment by tkobayas** [comment]
FDB: See https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/main/job/fdb/job/droolsjbpm-build-bootstrap-main.fdb/

"The Linux - Full Downstream Build" link on this page refers to the previous FDB (#734).

The current FDB (after the BanDuplicateClasses fix) is now running on #735. Let's wait to finish it.

**Comment by tkobayas** [comment]
Confirmed that the latest FDB is basically successful.

https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/main/job/fdb/job/droolsjbpm-build-bootstrap-main.fdb/735/consoleText

2 Test failures are not related to this PR (Tomcat)
```
[2023-12-21T16:14:24.377Z] [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 33.264 s <<< FAILURE! - in org.kie.bc.DeploymentIT
[2023-12-21T16:14:24.377Z] [ERROR] org.kie.bc.DeploymentIT  Time elapsed: 33.259 s  <<< ERROR!
[2023-12-21T16:14:24.377Z] org.jboss.arquillian.container.spi.client.container.DeploymentException:
[2023-12-21T16:14:24.377Z] Cannot deploy business-monitoring-webapp.war: {"WFLYCTL0062: Composite operation failed and was rolled back. Steps that failed:" => {"Operation step-1" => {"WFLYCTL0080: Failed services" => {"jboss.deployment.unit.\"business-monitoring-webapp.war\".WeldStartService" => "Failed to start service
[2023-12-21T16:14:24.377Z]     Caused by: org.jboss.weld.exceptions.DeploymentException: WELD-000049: Unable to invoke void org.kie.workbench.common.screens.archetype.mgmt.backend.service.ArchetypeServiceImpl.postConstruct() on org.kie.workbench.common.screens.archetype.mgmt.backend.service.ArchetypeServiceImpl@7b2860e
[2023-12-21T16:14:24.377Z]     Caused by: org.jboss.weld.exceptions.WeldException: WELD-000049: Unable to invoke void org.kie.workbench.common.screens.archetype.mgmt.backend.service.ArchetypeServiceImpl.postConstruct() on org.kie.workbench.common.screens.archetype.mgmt.backend.service.ArchetypeServiceImpl@7b2860e
[2023-12-21T16:14:24.377Z]     Caused by: java.lang.reflect.InvocationTargetException
[2023-12-21T16:14:24.377Z]     Caused by: org.uberfire.java.nio.file.FileSystemNotFoundException: No filesystem for uri (default://system/system) found."}}}}
```

```
[2023-12-21T16:43:41.281Z] [ERROR] Tests run: 3, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 152.272 s <<< FAILURE! - in org.kie.wb.selenium.ui.ProjectLibraryIntegrationTest
[2023-12-21T16:43:41.281Z] [ERROR] importAndBuildProjectFromStockRepository(org.kie.wb.selenium.ui.ProjectLibraryIntegrationTest)  Time elapsed: 38.38 s  <<< ERROR!
[2023-12-21T16:43:41.281Z] org.openqa.selenium.TimeoutException: Expected condition failed: waiting for presence of element located by: By.id: add-project-dropdown (tried for 15 second(s) with 500 milliseconds interval)
[2023-12-21T16:43:41.281Z]      at org.kie.wb.selenium.ui.ProjectLibraryIntegrationTest.lambda$importAndBuildProjectFromStockRepository$0(ProjectLibraryIntegrationTest.java:51)
[2023-12-21T16:43:41.281Z]      at org.kie.wb.selenium.ui.ProjectLibraryIntegrationTest.importBuildDeployAndCheckArtifact(ProjectLibraryIntegrationTest.java:68)
[2023-12-21T16:43:41.281Z]      at org.kie.wb.selenium.ui.ProjectLibraryIntegrationTest.importAndBuildProjectFromStockRepository(ProjectLibraryIntegrationTest.java:49)
[2023-12-21T16:43:41.281Z] Caused by: org.openqa.selenium.NoSuchElementException:
[2023-12-21T16:43:41.281Z] Cannot locate an element using By.id: add-project-dropdown
[2023-12-21T16:43:41.281Z] For documentation on this error, please visit: http://seleniumhq.org/exceptions/no_such_element.html
[2023-12-21T16:43:41.281Z] Build info: version: '3.11.0', revision: 'e59cfb3', time: '2018-03-11T20:26:55.152Z'
[2023-12-21T16:43:41.281Z] System info: host: 'eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com', ip: '127.0.0.1', os.name: 'Linux', os.arch: 'amd64', os.version: '3.10.0-1160.99.1.el7.x86_64', java.version: '11.0.15'
[2023-12-21T16:43:41.281Z] Driver info: driver.version: unknown
[2023-12-21T16:43:41.281Z]      at org.kie.wb.selenium.ui.ProjectLibraryIntegrationTest.lambda$importAndBuildProjectFromStockRepository$0(ProjectLibraryIntegrationTest.java:51)
[2023-12-21T16:43:41.281Z]      at org.kie.wb.selenium.ui.ProjectLibraryIntegrationTest.importBuildDeployAndCheckArtifact(ProjectLibraryIntegrationTest.java:68)
[2023-12-21T16:43:41.281Z]      at org.kie.wb.selenium.ui.ProjectLibraryIntegrationTest.importAndBuildProjectFromStockRepository(ProjectLibraryIntegrationTest.java:49)
```

**Comment by tkobayas** [comment]
@mariofusco please merge this with https://github.com/kiegroup/droolsjbpm-integration/pull/3007 , thanks!

**Pull request merged**